openlibrary/utils/open_syllabus_project.py

```diff
@@ -0,0 +1,108 @@
+import os
+import json
+import sqlite3
+import gzip
+
+
+# Define the SQLite database file path
+# Usually placed in root OL directory
+db_file = 'osp_totals.db'
+
+
+# Function to get the total based on OLID
+def get_total_by_olid(olid: str) -> int | None:
+    """
+    Retrieves the total number of times a book with the given Open Library ID (OLID) has been assigned in syllabi
+    from the Open Syllabus Project database.
+
+    Args:
+        olid (str): The Open Library ID (OLID) of the book to retrieve the total for.
+
+    Raises:
+        Exception: If there is an error querying the database.
+    """
+
+    olid_int = olid.replace("/works/", "").replace("OL", "").replace("W", "")
+    try:
+        # Connect to the SQLite database
+        conn = sqlite3.connect(db_file)
```

Comment:
Note: Ray raises some potential perf concerns:

- Is this going to slow down our indexing/full reindexing?
- Is opening a new sqlite connection every time we call this fn a good idea? Or should we keep a long-lived connection open?
- This is ~1,000,000 rows of data ; Since it's effectively just a map of int -> int, storing the whole thing in mem is also potentially an option and would use ~8mb (4bytes for each int). Is this something we want?

I'm not an expert on sqlite; Maybe we just time this function to see how long it takes? But I think the we should be fine. If we notice a perf issue we can easily turn this function off until we fix it!